src/app/core/guards/auth.guard.ts

```diff
@@ -18,7 +18,9 @@ export class AuthGuard implements CanActivate {
       if (isSessionValid && this.account.isLoggedIn()) {
         return true;
       } else {
-        this.account.refreshAccount();
+        if (isSessionValid !== this.account.isLoggedIn()) {
+          this.account.clear();
```

Comment:
Why do this only conditionally?

Reply:
The logic here is only to clear out the account if it's in the invalid state though I realize now it probably isn't necessary since they're logged out otherwise.

Reply:
Yep, that was what I was getting at! :slightly_smiling_face: It looks like `isLoggedIn` doesn't actually make a network call, which is good, but if we're gonna log them out might as well do so regardless, right?